compiler/lib/parse_bytecode.ml

```diff
@@ -2305,7 +2303,10 @@ and compile infos pc state instrs =
         let x, _ = State.accu state in
         let y = Var.fresh () in
 
-        ( (Let (y, Prim (Eq, [ Pc (Int n); Pv x ])), loc) :: instrs
+        ( ( Let
+              (y, Prim (Eq, [ Pc (Int (Targetint.of_int32_warning_on_overflow n)); Pv x ]))
+          , loc )
+          :: instrs
```

Comment:
If the integer argument of the BEQ opcode does not fit in an int31, the bytecode is not correctly compilable to Wasm. Isn’t a mere warning a bit weak in this case? If the user decides to ignore it, they could get a garbage program.

Reply:
I've changed it to raise an exception